src/Package/Impl/Shell/VisualComponentToolWindowAdapter.cs

```diff
@@ -31,6 +31,30 @@ public class VisualComponentToolWindowAdapter<T> : IVisualComponentContainer<T>
 
         private IVsWindowFrame VsWindowFrame => _vsWindowFrame ?? (_vsWindowFrame = _toolWindowPane.Frame as IVsWindowFrame);
 
+        public string CaptionText
+        {
+            get { return _toolWindowPane.Caption; }
+            set { _toolWindowPane.Caption = value; }
+        }
+
+        public string StatusText
+        {
+            get {
+                string text = string.Empty;
+                VsAppShell.Current.DispatchOnUIThread(() => {
```

Comment:
Oh yeah, I wasn't using that one.  I guess I'll change it to require the caller to switch to UI thread.  Let me know if there's a better way.  The places where I'm using the setter are already from UI thread.